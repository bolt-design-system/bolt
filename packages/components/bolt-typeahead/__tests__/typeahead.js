import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';
import { fixture as html } from '@open-wc/testing-helpers';
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

const timeout = 120000;

describe('<bolt-typeahead> Component', () => {
  let page;

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  afterAll(async () => {
    await stopTwigRenderer();
    await page.close();
  }, timeout);

  test('Typeahead Twig Template Renders', async () => {
    const results = await renderTwigString(`
      {% include "@bolt-components-form/form.twig" with {
        children: include("@bolt-components-typeahead/typeahead.twig", {
          attributes: {
            class: [
              "js-typeahead-hook"
            ]
          },
          max_results: 5,
        }),
        attributes: {
          action: "https://www.pega.com/search",
          target: "_blank",
          method: "GET"
        }
      } %}
    `);
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
