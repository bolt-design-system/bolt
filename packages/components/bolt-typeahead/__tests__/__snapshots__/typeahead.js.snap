// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-typeahead> Component Typeahead Twig Template Renders 1`] = `
<form action="https://www.pega.com/search"
      target="_blank"
      method="GET"
>
  <bolt-typeahead class="js-typeahead-hook"
                  max-results="5"
  >
    <div class="c-bolt-typeahead">
      <label for="12345"
             class="c-bolt-typeahead__label"
      >
        <span class="u-bolt-visuallyhidden">
        </span>
      </label>
      <input id="12345"
             type="text"
             autocomplete="off"
             aria-autocomplete="list"
             class="c-bolt-typeahead__input"
             placeholder="Enter your search query"
             value
             required
             name
      >
      <bolt-button size="medium"
                   color="text"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
                   icon-only
                   onclick="this.previousElementSibling.focus()"
                   class="c-bolt-typeahead__button c-bolt-typeahead__button--clear"
      >
        <button type="reset"
                class="c-bolt-button c-bolt-button--medium c-bolt-button--text c-bolt-button--border-radius-regular c-bolt-button--center c-bolt-button--icon-only"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__icon is-empty">
          </replace-with-children>
          <replace-with-children class="c-bolt-button__item">
            Clear search results
          </replace-with-children>
          <replace-with-children class="c-bolt-button__icon">
            <bolt-icon class="c-bolt-typeahead__icon"
                       name="close-solid"
            >
              <svg viewbox="0 0 24 24"
                   xmlns="http://www.w3.org/2000/svg"
              >
                <g fill="none"
                   fill-rule="evenodd"
                >
                  <path d="M0 0h24v24H0z">
                  </path>
                  <circle cx="11"
                          cy="11"
                          r="11"
                          transform="translate(1 1)"
                          fill="currentColor"
                  >
                  </circle>
                  <path d="M15.536 14.121l-2.122-2.12 2.122-2.122a1.003 1.003 0 000-1.414 1.002 1.002 0 00-1.415 0L12 10.586l-2.121-2.12a1 1 0 00-1.414 1.413L10.585 12l-2.12 2.121a1 1 0 001.414 1.414l2.12-2.12 2.122 2.12c.39.39 1.026.39 1.415 0a1.003 1.003 0 000-1.414"
                        fill="#FFF"
                  >
                  </path>
                </g>
              </svg>
            </bolt-icon>
          </replace-with-children>
        </button>
      </bolt-button>
      <div class="c-bolt-typeahead__button c-bolt-typeahead__button--submit">
        <bolt-button size="medium"
                     color="text"
                     width="auto"
                     border-radius="regular"
                     align="center"
                     transform="none"
                     icon-only
                     class="js-typeahead-hook"
        >
          <button type="submit"
                  class="c-bolt-button c-bolt-button--medium c-bolt-button--text c-bolt-button--border-radius-regular c-bolt-button--center c-bolt-button--icon-only"
                  is="shadow-root"
          >
            <replace-with-children class="c-bolt-button__icon is-empty">
            </replace-with-children>
            <replace-with-children class="c-bolt-button__item">
              Submit search query
            </replace-with-children>
            <replace-with-children class="c-bolt-button__icon">
              <bolt-icon class="c-bolt-typeahead__icon"
                         name="search"
              >
                <svg viewbox="0 0 24 24"
                     xmlns="http://www.w3.org/2000/svg"
                >
                  <g fill="none"
                     fill-rule="evenodd"
                  >
                    <path d="M0 0h24v24H0z">
                    </path>
                    <path d="M9.518 15.035A5.524 5.524 0 014 9.517 5.524 5.524 0 019.518 4a5.524 5.524 0 015.518 5.517 5.524 5.524 0 01-5.518 5.518m12.189 5.258l-6.221-6.222a7.469 7.469 0 001.55-4.554C17.036 5.372 13.664 2 9.518 2S2 5.372 2 9.517c0 4.145 3.372 7.518 7.518 7.518a7.471 7.471 0 004.554-1.55l6.22 6.222a1 1 0 001.415-1.414"
                          fill="currentColor"
                    >
                    </path>
                  </g>
                </svg>
              </bolt-icon>
            </replace-with-children>
          </button>
        </bolt-button>
      </div>
    </div>
  </bolt-typeahead>
</form>
`;
